src/integrationtests/java/com/aws/greengrass/integrationtests/ipc/VerifyClientDeviceIdentityTest.java

```diff
@@ -0,0 +1,181 @@
+import software.amazon.awssdk.aws.greengrass.VerifyClientDeviceIdentityResponseHandler;
+import software.amazon.awssdk.aws.greengrass.model.ClientDeviceCredential;
+import software.amazon.awssdk.aws.greengrass.model.UnauthorizedError;
+import software.amazon.awssdk.aws.greengrass.model.VerifyClientDeviceIdentityRequest;
+import software.amazon.awssdk.aws.greengrass.model.VerifyClientDeviceIdentityResponse;
+import software.amazon.awssdk.eventstreamrpc.EventStreamRPCConnection;
+import software.amazon.awssdk.services.greengrassv2data.GreengrassV2DataClient;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import static com.aws.greengrass.testcommons.testutilities.TestUtils.asyncAssertOnConsumer;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
+
+@ExtendWith({GGExtension.class, UniqueRootPathExtension.class, MockitoExtension.class})
+class VerifyClientDeviceIdentityTest {
+    private static GlobalStateChangeListener listener;
+    @TempDir
+    Path rootDir;
+    private Kernel kernel;
+    @Mock
+    private GreengrassServiceClientFactory clientFactory;
+    @Mock
+    private GreengrassV2DataClient client;
+    @Mock
+    private IotAuthClient iotAuthClient;
+
+    private static void verifyClientIdentity(GreengrassCoreIPCClient ipcClient,
+                                              VerifyClientDeviceIdentityRequest request,
+                                              Consumer<VerifyClientDeviceIdentityResponse> consumer) throws Exception {
+        VerifyClientDeviceIdentityResponseHandler handler = ipcClient.verifyClientDeviceIdentity(request, Optional.empty());
+        VerifyClientDeviceIdentityResponse response = handler.getResponse().get(10,TimeUnit.SECONDS);
+        consumer.accept(response);
+    }
+
+    @BeforeEach
+    void  beforeEach() {
+        // Set this property for kernel to scan its own classpath to find plugins
+        System.setProperty("aws.greengrass.scanSelfClasspath", "true");
+        kernel = new Kernel();
+        kernel.getContext().put(GreengrassServiceClientFactory.class, clientFactory);
+
+
+        when(clientFactory.getGreengrassV2DataClient()).thenReturn(client);
+        lenient().when(iotAuthClient.getActiveCertificateId("VALID PEM")).thenReturn(
+                Optional.of("SOME-CERT")
+        );
+        lenient().when(iotAuthClient.getActiveCertificateId("INVALID PEM")).thenReturn(
+                Optional.empty()
+        );
```

Comment:
setup these mocks in the individual test case so that you aren't using `lenient`

Have a test case which includes the auth client throwing an exception